components/Tabs/Tab.tsx

```diff
@@ -1,32 +1,43 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import Button from '../Button';
 import Context from '../../context';
 
 interface TabProps {
     id: String;
+    type: 'cap' | 'availability';
 }
 
 function Tab(props: TabProps): JSX.Element {
-    const { id } = props;
-
-    return (
-        <Context.Consumer>
-            {
-                (state: any) => (
-                    <li className="tab">
-                        <Button
-                            onClick={() => state.setGlobalState({ activeTab: id })}
-                            variant="primary"
-                            size="sm"
-                            className={id === state.activeTab ? 'primary' : 'inverted'
-                        }>
-                            {id}
-                        </Button>
-                    </li>
-                )
+    const { id, type } = props;
+    const { setGlobalState, filters: { 
+            activeTab, availabilityFilter
             }
-        </Context.Consumer>
-    )
+        } = useContext(Context);
+    const isActive = id === activeTab || id === availabilityFilter;
+    return (
+        <li className="tab">
+            <Button
+                onClick={() =>
+                    type === "cap"
+                        ? isActive
+                            ? setGlobalState({ filters:{availabilityFilter, activeTab: 'all'} })
+                            : setGlobalState({ filters:{availabilityFilter, activeTab: id} })
+                        : isActive
+                        ? setGlobalState({
+                              filters:{ activeTab ,availabilityFilter: 'none'},
+                          })
+                        : setGlobalState({
+                              filters:{ activeTab, availabilityFilter: id},
+                          })
+                }
```

Comment:
Same goes for this about logic, Maybe it's nicer to put this inside a more seperate function?

e.g. `function handleGBStatusFilter() {}`

Reply:
This is an insane nested ternary lol. I forgot to come back to refactor it